### PR TITLE
Generate translated files when Crowdin PRs are opened

### PR DESCRIPTION
**Pull request by gnardini**
This PR sets up Crowdin to send PRs with translations and generates files with the translated dApp Directory when new commits are made to it.

### CODE REVIEW

File: .github/workflows/file_generator.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  generate-translated-files:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '12'
```
Comment (silasbw) [inline .github/workflows/file_generator.yaml:18]: `node-version: '16'` (to match the other workflow)

File: scripts/processTranslations.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+const fs = require('fs')
```
Comment (silasbw) [inline scripts/processTranslations.js:1]: What's the reason this js (instead of ts)?

File: scripts/processTranslations.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+const i18next = require('i18next')
+
+const localizedFiles = fs.readdirSync('locales')
+const resources = localizedFiles.reduce((acc, file) => {
+  const lng = file.substring(0, file.length - '.json'.length)
```
Comment (silasbw) [inline scripts/processTranslations.js:6]: `path.parse(file).name`?

File: scripts/processTranslations.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
+const fs = require('fs')
+const i18next = require('i18next')
+
+const localizedFiles = fs.readdirSync('locales')
+const resources = localizedFiles.reduce((acc, file) => {
```
Comment (silasbw) [inline scripts/processTranslations.js:5]: naming nit: "file" -> "fileName"

File: scripts/processTranslations.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  const lng = file.substring(0, file.length - '.json'.length)
+  return {
+    ...acc,
+    [lng]: { translation: require(`../locales/${lng}.json`) },
```
Comment (silasbw) [inline scripts/processTranslations.js:9]: isn't this:

```
[lng]: { translation: require(`../locales/${file}`) },
```

(using `lng` to reconstruct `file` made me think there was something else going on)

File: scripts/processTranslations.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+function t(key, lng) {
+  return i18next.t(key, { lng })
+}
+
+async function generateTranslatedFiles() {
+  const dAppDirectory = JSON.parse(
+    fs.readFileSync('src/valora-dapp-list.json', 'utf8'),
+  )
+  for (const file of localizedFiles) {
+    const lng = file.substring(0, file.length - '.json'.length)
```
Comment (silasbw) [inline scripts/processTranslations.js:32]: use `path` like above

File: scripts/processTranslations.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    fs.writeFileSync(
+      `translations/valora-dapp-list-${file}`,
+      JSON.stringify(localizedDirectory, undefined, 2),
+    )
+  }
+}
+
+generateTranslatedFiles()
+  .then(() => console.log('Finished'))
+  .catch(console.error)
```
Comment (silasbw) [inline scripts/processTranslations.js:56]: exit with a non zero exit code if this failed (that's the UNIX convention for failure and will fail CI)

```
.catch(error => {
  console.error(error)
  process.exit(1)
})
```

File: package.json [hunk 8]
```diff
@@ -13,6 +13,7 @@
   },
   "prettier": "@valora/prettier-config",
   "devDependencies": {
+    "@actions/github": "^5.0.0",
```
Comment (silasbw) [inline package.json:16]: is this used in the CI action or something?  I wasn't sure where we were actually using it.

**Review by silasbw — APPROVED** [review]
nice :100:

File: scripts/processTranslations.js (hunk 2 above)
Comment (gnardini) [reply scripts/processTranslations.js:1]: I always have configuration issues trying to run typescript in standalone files like this, I'm trying now and running into errors, looking into how to fix them

File: package.json (hunk 8 above)
Comment (gnardini) [reply package.json:16]: I was, I tried using it but ended up going a different route, removing

File: scripts/processTranslations.js (hunk 2 above)
Comment (gnardini) [reply scripts/processTranslations.js]: Done! :)

File: .crowdin.yml [hunk 9]
```diff
@@ -0,0 +1,3 @@
+files:
+  - source: /locales/base.json
+    translation: /locales/%two_letters_code%.json
```
Comment (jeanregisser) [inline .crowdin.yml:3]: So it's recognized with the right type in Crowdin (allows pluralization, though of course it doesn't seem we need that here, but doesn't hurt).
```suggestion
    translation: /locales/%two_letters_code%.json
    type: i18next_json
```

**Review by jeanregisser — APPROVED** [review]
Great! 👍 

We should use the `valora-bot-crowdin` GH account in the Crowdin setup for this.
It has access to specific repos, to respect the principle of least privilege.
We can see this together on Monday.

**Pull request merged**